### PR TITLE
gh-104040: Silence a "comparison is always false" warning

### PR DESCRIPTION
**Pull request by TabLand**
On i686 by clarifying datatype used in check_ticks_per_second

* Issue: gh-104040

### CODE REVIEW

**Review by abalkin — APPROVED** [review]
LGTM

**Comment by TabLand** [comment]
Hi @pganssle,
Appreciate you're busy,
Would it be possible to review this when you get some extra bandwidth?

**Comment by TabLand** [comment]
Hi @abalkin / @arhadthedev,
Would it be possible to request a different 2nd reviewer?
Perhaps @vstinner?

File: Modules/timemodule.c [hunk 1]
```diff
@@ -69,7 +69,8 @@ check_ticks_per_second(long tps, const char *context)
 {
     /* Effectively, check that _PyTime_MulDiv(t, SEC_TO_NS, ticks_per_second)
        cannot overflow. */
-    if (tps >= 0 && (_PyTime_t)tps > _PyTime_MAX / SEC_TO_NS) {
+    _PyTime_t max_time_in_seconds = _PyTime_MAX / SEC_TO_NS;
+    if (tps >= 0 && (_PyTime_t)tps > max_time_in_seconds) {
```
Comment (vstinner) [inline Modules/timemodule.c:73]: I dislike this fix. You should use LONG_MAX and the preprocessor instead, something like:

```c
#ifdef LONG_MAX >= _PyTime_MAX
if (...)
#endif
```

The compiler is correct that with 32-bit long (which is the case on Windows 64-bit), the condition is always false:

```py
$ python
>>> SEC_TO_NS = 10**9
# _PyTime_t is always signed 64-bit
>>> _PyTime_MAX=2**63-1

# 64-bit signed long
>>> LONG_MAX=2**63-1
>>> LONG_MAX > (_PyTime_MAX // SEC_TO_NS)
True

# 32-bit long
>>> LONG_MAX=2**31-1
>>> LONG_MAX > (_PyTime_MAX // SEC_TO_NS)
False
```

**Comment by TabLand** [comment]
Hi @vstinner,

Apologies for the delay. 

It looks like the [recent reworking](https://github.com/python/cpython/commit/5c5022b8625e34f0035ad5a23bc4c2f16649d134) you've done on the timemodule.c file now makes this pull request obsolete.

Many thanks for your efforts 🙂️

**Comment by vstinner** [comment]
> It looks like the [recent reworking](https://github.com/python/cpython/commit/5c5022b8625e34f0035ad5a23bc4c2f16649d134) you've done on the timemodule.c file now makes this pull request obsolete.

Oh, right. I just removed the check. IMO it's no longer needed.

Thanks for your PR anyway.

**Pull request closed without merging**